### PR TITLE
create notes for each harbor scan, reference note name in occurrences

### PR DESCRIPTION
**Pull request by mrparkers**
there are a few places where the coverage could be improved.  I'll add in a few more tests tomorrow

### CODE REVIEW

File: listener/listener.go [hunk 1]
```diff
@@ -63,89 +63,130 @@ func (l *listener) ProcessEvent(w http.ResponseWriter, request *http.Request) {
 		log.Error("error reading webhook event", zap.Error(err))
 		return
 	}
+
+	// only process events that occur when scanning is finished
+	if event.Type != harbor.SCANNING_FAILED && event.Type != harbor.SCANNING_COMPLETED {
```
Comment (alexashley) [inline listener/listener.go:68]: Probably on this already, but now that we're only listening for these two events, we can drop [`PUSH_ARTIFACT` from the demo](https://github.com/rode/demo/blob/cc937553b24c950c1f1d7a0a4a84c3e957e9872b/tf/modules/harbor-config/main.tf#L9)

File: listener/listener.go [hunk 2]
```diff
@@ -63,89 +63,130 @@ func (l *listener) ProcessEvent(w http.ResponseWriter, request *http.Request) {
...
+}
+
+func (l *listener) processEventResource(event *harbor.Event, resource *harbor.Resource) error {
+	log := l.logger.Named("processEventResource").With(zap.Any("resource", resource))
+	log.Debug("processing event resource")
+
+	report := resource.ScanOverview.Report
+
+	// allow for one minute to process this event resource
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
```
Comment (alexashley) [inline listener/listener.go:101]: Does Harbor expect webhooks to return in a certain amount of time?

File: listener/listener.go [hunk 3]
```diff
@@ -175,16 +216,62 @@ func (l *listener) createVulnerabilityOccurrences(event *harbor.Event) ([]*grafe
...
+					Url:   artifactUrl,
+				},
+			},
+			Type: &grafeas_go_proto.Note_Discovery{
+				Discovery: &discovery_go_proto.Discovery{
+					AnalysisKind: common_go_proto.NoteKind_VULNERABILITY,
+				},
+			},
+		},
+		NoteId: fmt.Sprintf("harbor-scan-%s", report.ReportId),
```
Comment (alexashley) [inline listener/listener.go:242]: Another demo-related followup is that the sample policy we have for Harbor at the moment looks at the [exact note name](https://github.com/rode/demo/blob/cc937553b24c950c1f1d7a0a4a84c3e957e9872b/tf/modules/rode/policies/harbor.rego#L12), that'll have to change to `startswith` instead

**Review by alexashley — APPROVED** [review]
LGTM

File: listener/listener.go (hunk 2 above)
Comment (mrparkers) [reply listener/listener.go:101]: Just tested it, harbor doesn't seem to care if the response takes longer than a minute.

**Pull request merged**